core/psa_crypto_slot_management.c

```diff
@@ -660,7 +660,7 @@ psa_status_t psa_reserve_free_key_slot(psa_key_id_t *volatile_key_id,
             /* Refresh slot_idx, for when the slot is not the original
              * selected_slot but rather unused_persistent_key_slot.  */
             slot_idx = selected_slot - global_data.key_slots;
-            *volatile_key_id = PSA_KEY_ID_VOLATILE_MIN + slot_idx;
+            *volatile_key_id = PSA_KEY_ID_VOLATILE_MIN + (psa_key_id_t) slot_idx;
```

Comment:
As [discussed](https://github.com/Mbed-TLS/mbedtls/pull/10200#pullrequestreview-2937148566) on the back-port PR was this fix required ?